stakingModel/test/SimpleDAIStaking.test.js

```diff
@@ -90,6 +90,117 @@ contract("SimpleDAIStaking - staking with DAI mocks", ([founder, staker]) => {
+    const cdai1 = await cDAIMock.new(dai.address);
+    await dai.mint(cdai1.address, web3.utils.toWei("100", "ether"));
+    let balanceBefore = await cdai1.balanceOf(avatar.address);
+    let encodedCall = web3.eth.abi.encodeFunctionCall(
+      {
+        name: "recover",
+        type: "function",
+        inputs: [
+          {
+            type: "address",
+            name: "_token"
+          }
+        ]
+      },
+      [cdai1.address]
+    );
+    await controller.genericCall(simpleStaking.address, encodedCall, avatar.address, 0);
+    let balanceAfter = await cdai1.balanceOf(avatar.address);
+    expect(balanceAfter.toString()).to.be.equal(balanceBefore.toString());
+  });
+
+  it("should transfer funds when execute recover of token which the contract has some balance", async () => {
+    const cdai1 = await cDAIMock.new(dai.address);
+    await dai.mint(cdai1.address, web3.utils.toWei("100", "ether"));
+    const cdai1BalanceFounder = await cdai1.balanceOf(founder);
+    await cdai1.transfer(simpleStaking.address, cdai1BalanceFounder);
+    let balanceBefore = await cdai1.balanceOf(avatar.address);
+    let encodedCall = web3.eth.abi.encodeFunctionCall(
+      {
+        name: "recover",
+        type: "function",
+        inputs: [
+          {
+            type: "address",
+            name: "_token"
+          }
+        ]
+      },
+      [cdai1.address]
+    );
+    await controller.genericCall(simpleStaking.address, encodedCall, avatar.address, 0);
+    let balanceAfter = await cdai1.balanceOf(avatar.address);
+    expect(balanceAfter.sub(balanceBefore).toString()).to.be.equal(
+      cdai1BalanceFounder.toString()
+    );
+  });
+
+  it("should not transfer user's funds when execute recover", async () => {
+    await dai.mint(staker, web3.utils.toWei("100", "ether"));
+    await dai.approve(simpleStaking.address, web3.utils.toWei("100", "ether"), {
+      from: staker
+    });
+    let balanceBefore = await dai.balanceOf(avatar.address);
+    let stakerBalanceBefore = await dai.balanceOf(staker);
+    await simpleStaking
+      .stakeDAI(web3.utils.toWei("100", "ether"), {
+        from: staker
+      });
+    let encodedCall = web3.eth.abi.encodeFunctionCall(
+      {
+        name: "recover",
+        type: "function",
+        inputs: [
+          {
+            type: "address",
+            name: "_token"
+          }
+        ]
+      },
+      [dai.address]
+    );
+    await controller.genericCall(simpleStaking.address, encodedCall, avatar.address, 0);
+    await simpleStaking.withdrawStake({
+      from: staker
+    });
+    let balanceAfter = await dai.balanceOf(avatar.address);
+    let stakerBalanceAfter = await dai.balanceOf(staker);
+    expect(balanceAfter.toString()).to.be.equal(balanceBefore.toString());
+    expect(stakerBalanceAfter.toString()).to.be.equal(stakerBalanceBefore.toString());
+  });
+
+  it("should transfer excessive dai funds when execute recover", async () => {
```

Comment:
the test need to stake dai + have excessive funds
and check that after recover stake balance is still available

Reply:
@ordd

Reply:
updated
8f770b0e7739e63ca7fd86fbde237d5ae48653ee